ecs-init/docker/docker.go

```diff
@@ -173,10 +173,14 @@ func (c *Client) findAgentContainer() (string, error) {
 
 // StartAgent starts the Agent in Docker and returns the exit code from the container
 func (c *Client) StartAgent() (int, error) {
+	hostConfig, err := c.getHostConfig()
+	if err != nil {
+		return 0, err
```

Comment:
Should we have a different error code when the agent can't be configured? This failure mode seems different to a failure starting the container IMO.

Reply:
The other thing I'm wondering is should we log an error and continue? Is this worth blocking startup?